tests/phpunit/tests/block-bindings/wpBlockBindingsRegistry.php

```diff
@@ -321,4 +346,48 @@ public function test_is_registered_for_known_source() {
 		$result = $this->registry->is_registered( self::$test_source_name );
 		$this->assertTrue( $result );
 	}
+
+	/**
+	 * Test merging `uses_context` from multiple sources.
+	 *
+	 * @ticket 60525
+	 *
+	 * @covers ::register_block_bindings_source
+	 */
+	public function test_merging_uses_context_from_multiple_sources() {
+		$get_value_callback = function () {
+			return 'Anything';
+		};
+
+		$block_registry        = WP_Block_Type_Registry::get_instance();
+		$original_uses_context = $block_registry->get_registered( 'core/paragraph' )->uses_context;
+
+		register_block_bindings_source(
+			'test/source-one',
+			array(
+				'label'              => 'Test Source One',
+				'get_value_callback' => $get_value_callback,
+				'uses_context'       => array( 'commonContext', 'sourceOneContext' ),
+			)
+		);
+
+		register_block_bindings_source(
+			'test/source-two',
+			array(
+				'label'              => 'Test Source Two',
+				'get_value_callback' => $get_value_callback,
+				'uses_context'       => array( 'commonContext', 'sourceTwoContext' ),
+			)
+		);
+
+		$new_uses_context = $block_registry->get_registered( 'core/paragraph' )->uses_context;
+		// Check that the resulting `uses_context` contains the values from both sources.
+		$this->assertContains( 'commonContext', $new_uses_context );
+		$this->assertContains( 'sourceOneContext', $new_uses_context );
+		$this->assertContains( 'sourceTwoContext', $new_uses_context );
+		// Check that the resulting `uses_context` only added 3 items: commonContext, sourceOneContext, and sourceTwoContext.
+		$this->assertSame( count( $original_uses_context ) + 3, count( $new_uses_context ) );
+		// Check that the index is correct and it doesn't skip numbers after merging. Needed for the editor.
```

Comment:
Nit: how about the following comments?
```suggestion
		// Checks that the resulting `uses_context` contains the values from both sources.
		$this->assertContains( 'commonContext', $new_uses_context );
		$this->assertContains( 'sourceOneContext', $new_uses_context );
		$this->assertContains( 'sourceTwoContext', $new_uses_context );
		// Checks that the resulting `uses_context` added 3 unique items.
		$this->assertSame( count( $original_uses_context ) + 3, count( $new_uses_context ) );
		// Checks that the array isn't sparse to prevent issues in the editor.
```

I used 3rd person as in other comments.